### ISSUE TITLE
Feature: flatten union all plan to make profile plan straightforward

### ISSUE THREAD

**Issue by xudong963** [datafuselabs/databend#14682]
**Summary**
Currently
![img_v3_0287_37bb4c39-c27f-4563-b9ec-25f819a84a7g](https://github.com/datafuselabs/databend/assets/41979257/01435b1d-8885-4ed2-af70-e2db7a9518e5)

After 

![img_v3_0287_202d8d14-98ef-4502-b1c9-1ab07309063g](https://github.com/datafuselabs/databend/assets/41979257/47afa26c-d074-46b3-8328-01a3e26e90cd)

**Comment by sundy-li** [comment]
So does `union plan`

**Comment by xudong963** [comment]
> So does `union plan`

Yes, I will flatten union plan to make the profile plan flatten, not only to flatten the format tree.

**Comment by xudong963** [comment]
I'll continue the PR https://github.com/datafuselabs/databend/pull/14705 this week
